hello-proxy-impl/src/main/scala/com/example/helloproxy/impl/HelloProxyLoader.scala

```diff
@@ -30,13 +38,22 @@ abstract class HelloProxyApplication(context: LagomApplicationContext)
     with AhcWSComponents {
 
   private implicit val dispatcher: ExecutionContextExecutor = actorSystem.dispatcher
+  private implicit val sys: ActorSystem = actorSystem
+
+  def serviceDiscovery: SimpleServiceDiscovery
+
+  // See https://developer.lightbend.com/docs/akka-grpc/current/client/details.html#client-lifecycle
+  lazy val settings = GrpcClientSettings
+    .connectTo(GreeterService.name, serviceDiscovery)
+    .withServicePortName("https")
+    .withDeadline(Duration.create(5, TimeUnit.SECONDS)) // response timeout
+    .withConnectionAttempts(5) // use a small reconnectionAttempts value to cause a client reload in case of failure
+  lazy val clientConstructor = () => new GreeterServiceClient(settings)(materializer, dispatcher)
+  lazy val greeterServiceClient: RestartingClient[GreeterServiceClient] = new RestartingClient[GreeterServiceClient](clientConstructor)
```

Comment:
1. gRPC service lookup MUST filter by `portName` to only find services for `portName == "https"`
1. default deadline in akka-grpc is `Inf`. Set a deadline of `5.sec` (deadline is gRPC's name for remote request timeout)
1. instead of using a plain `GreeterServiceClient` I'm using a `RestartingClient[GreeterServiceClient]` so that when the downstream host fails the client will reconnect.